### DISCUSSION TITLE
Is there a way to dynamically show/hide tabs and score for a specific track (or even all tracks)?

### DISCUSSION THREAD

**Discussion by bali182** [CoderLine/alphaTab#1768] [Q&A] [answered]
I found in the docs that using alphatex this works well using  `\staff{score}` or `\staff{tabs}` or omitting `\staff` when I want both. 

Is it possible to dynamically change this when for example loading from a `.gp` file (for example dynamically hide or show tabs or score)? I found `Track#staves`, but this doesn't seem to be the correct approach:

```ts
const onRenderFinished = () => {
  for (const track of api.tracks) {
    for (const staff of track.staves) {
      // hide tab, show notation
      staff.showTablature = false
      staff.showStandardNotation = true
    }
  }
  api.renderFinished.off(onRenderFinished)
}
api.renderFinished.on(onRenderFinished)
```

**Comment by Danielku15** [comment, marked as answer]
You're almost on the right track. Changing the song model to specify which staves to show is correct. But the modification has to happen **before** the rendering. 

You can either hook into the [`scoreLoaded`](https://www.alphatab.net/docs/reference/api/scoreloaded) event or simply access at any time (e.g. as part of UI interactions) the [`score`](https://www.alphatab.net/docs/reference/api/score) and/or [`tracks`](https://www.alphatab.net/docs/reference/api/tracks) to do modifications. But at this point they just change the model, but there are not active listeners for any kind of modifications so you have to manually initiate an update.  

After the modification you can relayout and redraw the music notation with [`render`](https://www.alphatab.net/docs/reference/api/render/)

**Reply by bali182** [reply]
Fantastic, thank you!
